### PR TITLE
change 4bpp font format to allow dma2d rendering

### PR DESCRIPTION
**Pull request by TychoVrahe**
~~improves rendering speed, a small 4bpp font format change was necessary - swap pixel order inside byte a pad glyph width to always be even.~~

~~8bpp could support this too, but they are not used and we lack dma2d functions for utilizing that, so deferring that until needed.~~

~~also dma2d is now allowed in bootloader.~~

Format of the 4bpp font is changed to allow dma2d rendering in future (after drawing api is refactored).

### CODE REVIEW

**Comment by cepetr** [comment]
As I am working on the refactoring of the display/drawing API, it seems safer to focus only on changes related to the 4bpp fonts format—aligning rows to bytes and swapping nibbles. The newly refactored API for drawing will utilize DMA2D for all operations.

File: core/embed/lib/display.c [hunk 1]
```diff
@@ -291,63 +296,69 @@ void display_printf(const char *fmt, ...) {
 
 #endif  // TREZOR_PRINT_DISABLE
 
-#ifdef FRAMEBUFFER
+#if TREZOR_FONT_BPP == 4 && defined USE_DMA2D
 static void display_text_render(int x, int y, const char *text, int textlen,
```
[the remark this answers is not in the document]
Comment (TychoVrahe) [reply core/embed/lib/display.c]: fixed https://github.com/trezor/trezor-firmware/commit/60ad576006d550ed47636827936c59fa54639ebd

File: core/tools/codegen/gen_font.py [hunk 2]
```diff
@@ -47,10 +47,19 @@ def process_bitmap_buffer(buf: list[int], bpp: int) -> list[int]:
     elif bpp == 4:
         if len(res) % 2 > 0:
             res.append(0)
-        res = [
-            ((a & 0xF0) | (b >> 4))
-            for a, b in [res[i : i + 2] for i in range(0, len(res), 2)]
-        ]
+        width_adj = width
```
[the remark this answers is not in the document]
Comment (TychoVrahe) [reply core/tools/codegen/gen_font.py]: works too, removed the commented lines, they don't make much sense to me. and replaced y*width+width with (y+1)*width

https://github.com/trezor/trezor-firmware/commit/95cda62038a41a6968bb064ea0ac5d5f091addbb

**Comment by TychoVrahe** [comment]
reverted the dma2d related changes in https://github.com/trezor/trezor-firmware/commit/872f4b9c3f54b08e50eba8cd9838d2d02e17d172

set relative position for `gen_font` output in https://github.com/trezor/trezor-firmware/commit/a7b7f7ab48949ba98af5a1a6d28ca284adaa2a57 (+https://github.com/trezor/trezor-firmware/commit/6f78541c98f1c8ae344b2f4c0f35bc578484220b)

**Review by cepetr — APPROVED** [review]

**Pull request merged**